src/friction-compensation.cpp

```diff
@@ -0,0 +1,40 @@
+#include "simple-mpc/friction-compensation.hpp"
+
+namespace simple_mpc
+{
+
+  FrictionCompensation::FrictionCompensation(const Model & model, const bool with_free_flyer)
+  {
+    if (with_free_flyer)
+    {
+      // Ignore universe and root joints
+      nu_ = model.njoints - 2;
```

Comment:
Shouldn't it be `nu_ = model.nv` ?

Reply:
I'm not sure to understand why if the user ask **with** freeflyer set to **true** we should remove the freeflyer in the actuation size...

Reply:
> I'm not sure to understand why if the user ask **with** freeflyer set to **true** we should remove the freeflyer in the actuation size...

Ah I think I get what you are trying to do : You use the flag `with_freeflyer` to determine if the robot has a freeflyer or not and remove it if so.

You actually don't need a flag to accomplish that, you directly can use `model.joints[1]`.

What I had in mind with the flag s : if the robot has a freeflyer, does the user wants to remove it from the friction estimation or not ?